homeassistant/components/ohme/coordinator.py

```diff
@@ -0,0 +1,64 @@
+"""Ohme coordinators."""
+
+from datetime import timedelta
+import logging
+
+from ohme import ApiException
+
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_INTERVAL_ADVANCED, DEFAULT_INTERVAL_CHARGESESSIONS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OhmeChargeSessionsCoordinator(DataUpdateCoordinator):
+    """Coordinator to pull main charge state and power/current draw."""
+
+    def __init__(self, hass, config_entry):
+        """Initialise coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="Ohme Charge Sessions",
+            update_interval=timedelta(
+                minutes=config_entry.options.get(
+                    "interval_chargesessions", DEFAULT_INTERVAL_CHARGESESSIONS
```

Comment:
we don't allow setting the update interval through the Options. Since this is the case, I'm really thinking if it makes sense at all to use different coordinators, as you could just refresh the advanced part on every second "normal" refresh and it would also make the rest of the code easier.

Reply:
I think this approach makes a lot of sense. With the custom integration version, having 4 coordinators and some entities needing to handle multiple is a pain!